test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/integration/ProcessInstanceMigrationExternalDatabaseIntegrationTest.java

```diff
@@ -0,0 +1,210 @@
+        }
+
+        // Then
+        List<ProcessInstance> instances = processServicesClient.findProcessInstances(OLD_CONTAINER_ID, 0, 10);
+        Assertions.assertThat(instances).hasSize(1);
+        Assertions.assertThat(instances.get(0).getId()).isIn(instancesIds);
+
+        instances = processServicesClient.findProcessInstances(NEW_CONTAINER_ID, 0, 10);
+        Assertions.assertThat(instances).hasSize(1);
+        Assertions.assertThat(instances.get(0).getId()).isIn(instancesIds);
+    }
+
+    private Migration createMigration() throws IOException, JAXBException {
+        Plan plan = createPlan();
+        MigrationDefinition migrationDefinition = new MigrationDefinition();
+        migrationDefinition.setPlanId(plan.getId());
+        migrationDefinition.setKieServerId(kieServerId);
+        migrationDefinition.setProcessInstanceIds(Arrays.asList(instancesIds.get(0)));
+        migrationDefinition.setExecution(new Execution().setType(Execution.ExecutionType.SYNC));
+
+        HttpPost post = preparePost("/migrations");
+
+        StringWriter writer = new StringWriter();
+        mapper.writeValue(writer, migrationDefinition);
+        post.setEntity(new StringEntity(writer.toString()));
+        HttpResponse r = client.execute(post);
+        Assertions.assertThat(r.getStatusLine().getStatusCode()).isEqualTo(HttpStatus.SC_OK);
+        StringReader reader = new StringReader(EntityUtils.toString(r.getEntity()));
+        Migration migration = mapper.readValue(reader, Migration.class);
+        Assertions.assertThat(migration).isNotNull();
+        //Assertions.assertThat(migration.getDefinition().getRequester()).isEqualTo("kermit");
+        Assertions.assertThat(migration.getDefinition().getRequester()).isEqualTo("ANONYMOUS");
+        return migration;
+    }
+
+    private Plan createPlan() throws IOException, JAXBException {
+        Plan plan = new Plan();
+        plan.setSource(new ProcessRef().setContainerId(OLD_CONTAINER_ID).setProcessId(PROCESS_ID));
+        plan.setTarget(new ProcessRef().setContainerId(NEW_CONTAINER_ID).setProcessId(PROCESS_ID));
+
+        HttpPost post = preparePost("/plans");
+
+        StringWriter writer = new StringWriter();
+        mapper.writeValue(writer, plan);
+        post.setEntity(new StringEntity(writer.toString()));
+        HttpResponse r = client.execute(post);
+        Assertions.assertThat(r.getStatusLine().getStatusCode()).isEqualTo(HttpStatus.SC_OK);
+        StringReader reader = new StringReader(EntityUtils.toString(r.getEntity()));
+        return mapper.readValue(reader, Plan.class);
+    }
+
+    private HttpPost preparePost(String path) {
+        HttpPost post = new HttpPost(deploymentScenario.getProcessMigrationDeployment().getUrl()+"/rest" + path);
+        post.setHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON);
+        post.setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON);
+        return post;
```

Comment:
For me, this is too low level. Maybe we could follow the same we have for kieserver or workbench: adding a ProcessInstanceClient with a method createPlan(oldContainer, newContainer).
What do you think?

Reply:
I don't think that a new client is needed (If would be needed, there would be already created one in droolsjbpm-integration repo). For PIM service there should be mainly used the UI part.